Packages/com.unity.inputsystem/InputSystem/Plugins/Linux/SDLDeviceBuilder.cs

```diff
@@ -0,0 +1,329 @@
+                .WithByteOffset(4)
+                .WithSizeInBits((uint)yFeature.size * 8);
+
+            builder.AddControl(hatName + "/down")
+                .WithFormat(InputStateBlock.kTypeInt)
+                .WithLayout("Button")
+                .WithParameters("scale,scaleFactor=2147483647,clamp,clampMin=0,clampMax=1")
+                .WithByteOffset(4)
+                .WithSizeInBits((uint)yFeature.size * 8);
+
+            builder.AddControl(hatName + "/left")
+                .WithFormat(InputStateBlock.kTypeInt)
+                .WithLayout("Button")
+                .WithParameters("scale,scaleFactor=2147483647,clamp,clampMin=-1,clampMax=0,invert")
+                .WithByteOffset(0)
+                .WithSizeInBits((uint)xFeature.size * 8);
+
+            builder.AddControl(hatName + "/right")
+                .WithFormat(InputStateBlock.kTypeInt)
+                .WithLayout("Button")
+                .WithParameters("scale,scaleFactor=2147483647,clamp,clampMin=0,clampMax=1")
+                .WithByteOffset(0)
+                .WithSizeInBits((uint)xFeature.size * 8);
+
+            builder.AddControl(hatName + "/x")
+                .WithFormat(InputStateBlock.kTypeInt)
+                .WithLayout("Analog")
+                .WithParameters("scale,scaleFactor=2147483647,clamp,clampMin=0,clampMax=1")
+                .WithByteOffset(0)
+                .WithSizeInBits((uint)xFeature.size * 8);
+
+            builder.AddControl(hatName + "/y")
+                .WithFormat(InputStateBlock.kTypeInt)
+                .WithLayout("Analog")
+                .WithParameters("scale,scaleFactor=2147483647,clamp,clampMin=-1,clampMax=1,invert")
+                .WithByteOffset(4)
+                .WithSizeInBits((uint)yFeature.size * 8);
+        }
+
+        internal InputControlLayout Build()
+        {
+            var builder = new InputControlLayout.Builder
+            {
+                stateFormat = new FourCC('L', 'J', 'O', 'Y'),
+                extendsLayout = parentLayout
+            };
+
+            for (var i = 0; i < descriptor.controls.Count; i++)
+            {
+                SDLFeatureDescriptor feature = descriptor.controls[i];
+                switch (feature.featureType)
+                {
+                    case JoystickFeatureType.Axis:
+                    {
+                        SDLAxisUsage usage = (SDLAxisUsage)feature.usageHint;
+                        string featureName = SDLSupport.GetAxisNameFromUsage(usage);
+                        string parameters = "scale,scaleFactor=65538,clamp,clampMin=-1,clampMax=1";
+
+                        if (IsAxisX(feature) && i + 1 < descriptor.controls.Count)
+                        {
+                            SDLFeatureDescriptor nextFeature = descriptor.controls[i + 1];
+                            if (IsAxisY(nextFeature))
+                                BuildStickFeature(ref builder, feature, nextFeature);
```

Comment:
This seems to build a stick *plus* X and a Y controls at the same level. That doesn't seem useful given sticks already have X and Y axes.